From 0bbcb55cb082bde93dd60073629be1bfa4f3d739 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Tue, 28 Nov 2017 21:43:18 +0800
Subject: [PATCH 2928/5242] MLK-16604-1 usb: host: xhci-plat: add
 XHCI_MISSING_CAS quirk

commit  d57334e28d1d27850ea0ad318bb555ecad3c3c2c from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX8MQ USB3 host needs XHCI_MISSING_CAS quirk to warm reset the port to
enum the USB3 device plugged in while system sleep, as the port state is
stuck in polling mode after resume.

Signed-off-by: Li Jun <jun.li@nxp.com>
Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/host/xhci-plat.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 0500af6..6bc131d 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -285,6 +285,9 @@ static int xhci_plat_probe(struct platform_device *pdev)
 		if (device_property_read_bool(tmpdev, "usb3-lpm-capable"))
 			xhci->quirks |= XHCI_LPM_SUPPORT;
 
+		if (device_property_read_bool(sysdev, "usb3-resume-missing-cas"))
+			xhci->quirks |= XHCI_MISSING_CAS;
+
 		if (device_property_read_bool(tmpdev, "quirk-broken-port-ped"))
 			xhci->quirks |= XHCI_BROKEN_PORT_PED;
 
-- 
1.7.9.5

